src/coreclr/src/tools/Common/TypeSystem/Common/ThrowHelper.cs

```diff
@@ -42,6 +42,12 @@ public static void ThrowInvalidProgramException()
             throw new TypeSystemException.InvalidProgramException();
         }
 
+        [System.Diagnostics.DebuggerHidden]
+        public static void ThrowInvalidProgramException(ExceptionStringID id)
```

Comment:
Nit: the existing overload that also includes the method with the problem seems to be better, diagnostic-wise.